### PR TITLE
[bitnami/mongodb] Fix service.nameOverride and TLS certificate generation in standalone mode

### LINKED ISSUE #14353: [MongoDB] Standalone Architecture generates TLS certificates with incorrect hostnames
### Name and Version

bitnami/mongodb 13.6.2

### What steps will reproduce the bug?

1. Enable TLS with the "standalone" architecture
2. Connect to the server using the FQDN (*.svc.cluster.local) for the pod
3. Observe hostname mismatch

### Are you using any custom parameters or values?

```
--set tls.enabled=true
--set useStatefulSet=true # Unsure if this is relevant, but its what I was using
```

### What is the expected behavior?

The generated TLS certificate to be valid for the hostnames the pod is reachable at.

### What do you see instead?

The TLS certificates appear to be valid for a fictitious "mongodb-headless" service which does not exist, and is not created by the chart.

### Additional information

The culprit appears to be [this line](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/standalone/dep-sts.yaml#L154) (compare with [here](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/standalone/svc.yaml#L5)). Either the service.nameOverride should be used to set the service name, or the argument the certificate generation script would be the output of the "mongodb.fullname" helper to match the service name. From looking at the replicaset StatefulSet, the same issue appears to be present, but I have not tested this.

I'm happy to open a PR if you inform me of which of these fixes is the correct one. Using the `service.nameOverride` for the service name "feels" correct, but this could result in breaking changes.

### PR DESCRIPTION
**Pull request by migruiz4**
Signed-off-by: Miguel Ruiz <miruiz@vmware.com>

### Description of the change

Fixes an issue where `service.nameOverride` did not apply when running MongoDB in standalone mode.
This issue also affected certificate generation, as no headless service is generated.

### Applicable issues

  - fixes #14353

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/mongodb/templates/common-scripts-cm.yaml [hunk 1]
```diff
@@ -71,7 +71,11 @@ data:
     [alt_names]
     DNS.1 = $svc
     DNS.2 = $my_hostname
+    {{- if eq .Values.architecture "replicaset" }}
     DNS.3 = $my_hostname.$svc.$MY_POD_NAMESPACE.svc.cluster.local
+    {{- else }}
+    DNS.3 = $svc.$MY_POD_NAMESPACE.svc.cluster.local
+    {{- end }}
```
Comment (migruiz4) [inline bitnami/mongodb/templates/common-scripts-cm.yaml:78]: No FDQN is required in standalone mode, as no headless service is generated.

I replaced DNS.3 with the long service name.

File: bitnami/mongodb/templates/NOTES.txt [hunk 2]
```diff
@@ -187,7 +187,7 @@ To connect to your database from outside the cluster execute the following comma
 
 To access the MongoDB&reg; Prometheus metrics, get the MongoDB&reg; Prometheus URL by running:
 
-    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ include "mongodb.fullname" . }}-metrics {{ .Values.metrics.service.ports.metrics }}:{{ .Values.metrics.service.ports.metrics }} &
+    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ include "mongodb.service.nameOverride" . }}-metrics {{ .Values.metrics.service.ports.metrics }}:{{ .Values.metrics.service.ports.metrics }} &
```
Comment (fmulero) [inline bitnami/mongodb/templates/NOTES.txt:190]: [metrics-svc.yml](https://github.com/bitnami/charts/blob/main/bitnami/mongodb/templates/metrics-svc.yaml#L5) uses  `mongodb.fullname` template for name.

**Review by fmulero — CHANGES_REQUESTED** [review]
Great @migruiz4 Thanks a lot! I left a comment, Could you review it?

File: bitnami/mongodb/templates/NOTES.txt (hunk 2 above)
Comment (migruiz4) [reply bitnami/mongodb/templates/NOTES.txt:190]: I didn't notice the `-metrics`, nice catch!

**Review by fmulero — APPROVED** [review]
Great! LGTM

**Pull request merged**